### PR TITLE
docs/release-process.md: merge Kamil's checklist

### PR DESCRIPTION
**Pull request by kdomanski**
```release-note
NONE
```

### CODE REVIEW

File: docs/release-process.md [hunk 1]
```diff
@@ -17,20 +17,29 @@ to creating a new release. All these tests must be executed with every supported
...
-1. Default branch for https://github.com/kubermatic/kubermatic-installer has been set to the new branch
-1. The chart sync has been validated to be working (https://github.com/kubermatic/kubermatic-installer contains latest versions of helm charts from https://github.com/kubermatic/kubermatic)
-1. Now the new release can be created via https://github.com/kubermatic/kubermatic/releases. 
-  - Make sure to use the release branch to create the tag
-  - Drone will publish the release(Docker images & charts) 
-1. After the release has been created, the changelog must be updated. (Responsible: @kdomanski) 
+1. Kubernetes lifecycle
+    - Ensure ConfigMap for new kubelet version is added
+    - Ensure RBAC for that ConfigMap is added
+    - Ensure end-of-life Kubernetes has been disabled
```
Comment (alvaroaleman) [inline docs/release-process.md:23]: Might be worth adding "wait for new versions of external cloud controller managers and add them"? Else LBs and PVs wont work

**Comment by xrstf** [comment]
Can this list also contain a `go throught TODO comments and remove outdated code`?

**Comment by xrstf** [comment]
@kdomanski does this still match reality and can be merged?

**Comment by kdomanski** [comment]
Yes.

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**